src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1796,42 +1796,26 @@ private BaseListSyntax ParseBaseList()
             try
             {
                 // first type
```

Comment:
This is parsed properly.  That's because ParseType will not accept "where :" as a type name.  I'm not sure if we have tests for this though, so i will add them.